app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,54 @@
+class MarketsController < ApplicationController
+  def index
+    @markets = Market.order(market_name: :asc)
+    @all_markets = Market.order(market_name: :asc)
+  end
+
+  def show
+    @market = Market.find(params[:id])
+    @vendors = Vendor.where(market_id: params[:id])
+    render :show
+  end
+
+  def lettermarket
+    @markets = Market.where("market_name LIKE :prefix", prefix: "#{params[:letter]}%")
+    @all_markets = Market.order(market_name: :asc)
```

Comment:
It doesn't seem like you show all of the markets in the markest/1/:letter route, and if not you should take out the @all_markets variable. Again, in a big app unnecessary queries can affect performance.